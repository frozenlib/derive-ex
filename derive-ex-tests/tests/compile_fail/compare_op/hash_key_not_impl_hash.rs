use derive_ex::derive_ex;

#[derive(Eq, PartialEq)]
struct NoHash;

#[derive_ex(Hash)]
struct X {
    #[hash(key = NoHash)]
    x: u8,
}

fn main() {}
