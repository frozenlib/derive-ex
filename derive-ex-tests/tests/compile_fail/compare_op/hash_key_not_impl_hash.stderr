error[E0277]: the trait bound `NoHash: Hash` is not satisfied
 --> tests/compile_fail/compare_op/hash_key_not_impl_hash.rs:8:18
  |
8 |     #[hash(key = NoHash)]
  |                  ^^^^^^ the trait `Hash` is not implemented for `NoHash`
  |
help: consider annotating `NoHash` with `#[derive(Hash)]`
  |
4 + #[derive(Hash)]
5 | struct NoHash;
  |
